### PR TITLE
[WIP] Support knowledge distillation

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR supports knowledge distillation training for FastSpeech.

Now under construction.

### CODE REVIEW

File: egs/ljspeech/tts1/run.sh [hunk 1]
```diff
@@ -228,3 +228,91 @@ if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ]; then
...
+                --model ${expdir}/results/${model} \
+                --save-durations true \
+                --config ${decode_config}
+        # concatenate scp files
+        for n in $(seq ${nj}); do
+            cat "${outdir}/${name}/feats.$n.scp" || exit 1;
+        done > ${outdir}/${name}/feats.scp
+        for n in $(seq ${nj}); do
+            cat "${outdir}/${name}/durations.$n.scp" || exit 1;
+        done > ${outdir}/${name}/durations.scp
```
Comment (r9y9) [inline egs/ljspeech/tts1/run.sh]: We may want to add logic to remove bad samples (e.g. contains deletions and repetitions). Possible solutions:
1) Check focus rate to detect the repeat or deletion (probably easier)
2) Use ASR-based objective evaluation (i.e. use utterances whose CER=0.0) (code will be dirty...)

Comment (kan-bayashi) [reply egs/ljspeech/tts1/run.sh]: Currently, I removed the samples reaching the maximum length.

**Comment by kan-bayashi** [comment]
## FastSpeech v2 @ 200 epoch
スクリーンショット 2019-10-19 午前11 01 23

## FastSpeech v4 @ 200 epoch
スクリーンショット 2019-10-19 午前11 01 47

It seems nice.

## v4 Training curve 
スクリーンショット 2019-10-19 午前11 04 36

Overfitting is not serious compared to the v1 ~ v3.

**Comment by kan-bayashi** [comment]
I finished training.
Here are the samples.
https://drive.google.com/open?id=1J4OLej_ZIarGqN9pqyp9Kzn-KnXCOjVd

**Comment by kan-bayashi** [comment]
I revisit this PR from the end of this week.

**Pull request closed without merging**